collated/test/A0127014W.md

```diff
@@ -0,0 +1,258 @@
+# A0127014W
+###### \java\guitests\CollapseExpandCommandTest.java
+``` java
+package guitests;
+
+import static org.junit.Assert.*;
+
+import org.junit.Test;
+
+public class CollapseExpandCommandTest extends AddressBookGuiTest{
+
+    @Test
+    public void collapse_alreadyCollapsed() {
+        commandBox.runCommand("collapse");
+        assertResultMessage("Task view collapsed");
+        commandBox.runCommand("collapse");
+        assertResultMessage("Already in collapsed view, type \"expand\" to go into expanded view");
+    }
+
+    @Test
+    public void collapse_after_expand() {
+        commandBox.runCommand("expand");
+        commandBox.runCommand("collapse");
+        assertResultMessage("Task view collapsed");
+    }
+
+    /*
+    @Test
+    public void expand_alreadyExpanded() {
+        commandBox.runCommand("expand");
+        assertResultMessage("Task view expanded");
+        commandBox.runCommand("expand");
+        assertResultMessage("Already in expanded view, type \"collapse\" to go into collapsed view");
+    }
+
+    @Test
+    public void expand_after_collapse() {
+        commandBox.runCommand("collapse");
+        commandBox.runCommand("expand");
+        assertResultMessage("Task view expanded");
+    }
+    */
+}
+```
+###### \java\guitests\SelectCommandTest.java
+``` java
+    private void assertSelectionInvalid(int index) {
+        commandBox.runCommand("select " + index);
+        if(taskListPanel.getNumberOfTasks() < 1){
+            assertResultMessage("The task index provided is invalid" + "\n" + "Can't select from an empty list");
+        }
+        else{
+            assertResultMessage("The task index provided is invalid" + "\n" + "Valid index range: 1 to " + taskListPanel.getNumberOfTasks());
+        }
+    }
+```
+###### \java\teamfour\tasc\logic\CommandHelperTest.java
+``` java
+    @Test
+    public void convertStringToMultipleDates_shortName_correct_date_month_year() {
+        String date = "13 sep 2013";
+        List<Date> dates = CommandHelper.convertStringToMultipleDates(date);
+        assertTrue(dates.get(0).getDate() == 13);
+        assertTrue(dates.get(0).getMonth() == 8);
+        assertTrue(dates.get(0).getYear() == 2013 - 1900);
+
+        String date2 = "13th sept 2000";
+        List<Date> dates2 = CommandHelper.convertStringToMultipleDates(date2);
+        assertTrue(dates2.get(0).getDate() == 13);
+        assertTrue(dates2.get(0).getMonth() == 8);
+        assertTrue(dates2.get(0).getYear() == 2000 - 1900);
+
+        String date3 = "sept 13 1900";
+        List<Date> dates3 = CommandHelper.convertStringToMultipleDates(date3);
+        assertTrue(dates3.get(0).getDate() == 13);
```

Comment:
consider using a setup method before the test to create these objects;  try to use common objects across tests to reduce code duplication